packages/api/src/routes/medical/fhir-r4-proxy.ts

```diff
@@ -17,6 +35,53 @@ const fhirRouter = (fhirServerUrl: string) =>
       log(`Proxying to FHIR server: ${updatedURL}`);
       return updatedURL;
     },
+    //eslint-disable-next-line @typescript-eslint/no-unused-vars
+    userResDecorator: function (proxyRes, proxyResData, userReq, userRes) {
+      const data = JSON.parse(proxyResData.toString("utf8"));
+      // v1 handle only known use case `/fhir/R4/{resource}`
+      if (data["resourceType"] === "Bundle" && "type" in data && data["type"] === "searchset") {
+        if ("link" in data) {
+          data["link"].forEach((link: fhirLink) => {
+            const defaultPaginationQueryParams = {
+              _getpages: data["id"],
+              _getpagesoffset: "0",
+              _count: "20",
+              _pretty: "true",
+              _bundletype: "searchset",
+            };
+            let queryParams = link.url.includes("?")
+              ? link.url
+                  .split("?")[1]
+                  .split("&")
+                  .reduce((a, v) => ({ ...a, [v.split("=")[0]]: v.split("=")[1] }), {})
+              : {};
+            queryParams = {
+              ...defaultPaginationQueryParams,
+              ...queryParams,
+            };
+            link.url =
+              Config.getApiUrl() +
+              userReq.baseUrl +
+              "?" +
+              Object.entries(queryParams)
+                .map(entry => `${entry[0]}=${entry[1]}`)
+                .join("&");
+          });
+        }
+        if ("entry" in data) {
+          data["entry"].forEach((entry: fhirEntry) => {
+            entry.fullUrl =
+              Config.getApiUrl() +
+              userReq.baseUrl +
+              "/" +
+              entry.resource.resourceType +
+              "/" +
+              entry.resource.id;
```

Comment:
template literals would make this easier to read - ie `${Config.getApiUrl()}${userReq.baseUrl}...`